test/sql/copy.sql.in

```diff
@@ -111,3 +111,458 @@ COPY hyper TO STDOUT DELIMITER ',';
+SELECT create_hypertable('table_with_chunk_trigger', 'time', chunk_time_interval => 1, create_default_indexes => false);
+
+-- Insert data to create all missing chunks
+\copy table_with_chunk_trigger from data/copy_data.csv with csv header;
+SELECT count(*) FROM table_with_chunk_trigger;
+
+-- Chunk 1: 1-2, Chunk 2: 2-3, Chunk 3: 3-4, Chunk 4: 4-5
+SELECT chunk_schema, chunk_name FROM timescaledb_information.chunks
+    WHERE hypertable_name = 'table_with_chunk_trigger' AND range_end_integer=5 \gset
+
+-- Create after trigger
+CREATE TRIGGER table_with_chunk_trigger_after_trigger
+    AFTER INSERT ON :chunk_schema.:chunk_name
+    FOR EACH ROW EXECUTE FUNCTION count_test_chunk_trigger();
+
+\copy table_with_chunk_trigger from data/copy_data.csv with csv header;
+SELECT count(*) FROM table_with_chunk_trigger;
+
+----------------------------------------------------------------
+-- Testing multi-buffer optimization
+-- (Hypertable without before insert trigger)
+----------------------------------------------------------------
+CREATE TABLE "table_without_bf_trigger" (
+    "time" bigint NOT NULL,
+    "value" double precision NOT NULL
+);
+
+SELECT create_hypertable('table_without_bf_trigger', 'time', chunk_time_interval => 1);
+
+-- Drop the default insert block trigger
+DROP TRIGGER ts_insert_blocker ON table_without_bf_trigger;
+
+\copy table_without_bf_trigger from data/copy_data.csv with csv header;
+
+SET client_min_messages TO DEBUG1;
+\copy table_without_bf_trigger from data/copy_data.csv with csv header;
+RESET client_min_messages;
+
+SELECT count(*) FROM table_without_bf_trigger;
+
+-- After trigger (CIM_MULTI_CONDITIONAL should be used)
+CREATE TRIGGER table_with_chunk_trigger_after_trigger
+    AFTER INSERT ON table_without_bf_trigger
+    FOR EACH ROW EXECUTE FUNCTION empty_test_trigger();
+
+SET client_min_messages TO DEBUG1;
+\copy table_without_bf_trigger from data/copy_data.csv with csv header;
+RESET client_min_messages;
+
+SELECT count(*) FROM table_without_bf_trigger;
+
+----------------------------------------------------------------
+-- Testing multi-buffer optimization
+-- (Chunks with different layouts)
+----------------------------------------------------------------
+CREATE TABLE "table_with_layout_change" (
+    "time" bigint NOT NULL,
+    "value1" double precision NOT NULL default 1,
+    "value2" double precision NOT NULL default 5
+);
+
+SELECT create_hypertable('table_with_layout_change', 'time', chunk_time_interval => 1);
+
+-- Create chunks with current layout
+\copy table_with_layout_change (time, value1) from data/copy_data.csv with csv header;
+SELECT * FROM table_with_layout_change ORDER BY time, value1, value2 LIMIT 10;
+
+-- Drop one column from hypertable
+ALTER TABLE table_with_layout_change DROP COLUMN value2;
+SELECT * FROM table_with_layout_change ORDER BY time, value1 LIMIT 10;
+
+-- Reexecute COPY on changed layout
+\copy table_with_layout_change from data/copy_data.csv with csv header;
```

Comment:
ok no layout change is actually change happening here since we copy the same csv we used before so no new chunk is created. Since we only add they would not result in different physical layouts